class-plugin-autoupdate-filter.php

```diff
@@ -146,24 +146,47 @@ public function filter_maybe_disable_all_autoupdates( $update, $item ): bool {
 	 * @return bool True to update, false to not update.
 	 */
 	public function filter_enforce_delay( $update, $item ): bool {
+		// protect against non-bool being returned from this function
+		if ( null === $update ) {
+			$update = false;
+		}
+
 		// no delay if site is a canary site
 		$site_url = wp_parse_url( home_url(), PHP_URL_HOST );
 		if ( isset( $this->settings->canary_sites ) && in_array( $site_url, $this->settings->canary_sites, true ) ) {
 			return $update;
 		}
 
-		// otherwise add delay to plugin updates
-		if ( true === $update ) {
-			$helpers = new Plugin_Autoupdate_Filter_Helpers();
+		// otherwise apply delay logic
+		$helpers = new Plugin_Autoupdate_Filter_Helpers();
+
+		$plugin_file        = empty( $item->plugin ) ? '' : $item->plugin;
+		$plugin_slug        = empty( $item->slug ) ? '' : $item->slug;
+		$plugin_new_version = empty( $item->new_version ) ? '0.0.0' : $item->new_version;
 
-			$plugin_file        = empty( $item->plugin ) ? '' : $item->plugin;
-			$plugin_slug        = empty( $item->slug ) ? '' : $item->slug;
-			$plugin_new_version = empty( $item->new_version ) ? '0.0.0' : $item->new_version;
+		$has_delay_passed = $helpers->has_delay_passed( $plugin_slug, $plugin_new_version, $plugin_file );
 
-			$has_delay_passed = $helpers->has_delay_passed( $plugin_slug, $plugin_new_version, $plugin_file );
-			if ( false === $has_delay_passed ) {
-				$update = false;
+		if ( false === $has_delay_passed ) {
+			$option_key = 'plugin_update_delays';
+			$delays     = get_option( $option_key, array() );
+			if ( isset( $delays[ $plugin_file ][ $plugin_new_version ] ) ) {
+				$delay_date = $delays[ $plugin_file ][ $plugin_new_version ];
```

Comment:
I think it's worth checking `is_numeric( $delay_date )` before using it on `date_i18n()`

Reply:
Thanks! Added.